packages/flutter/lib/src/widgets/layout_builder.dart

```diff
@@ -269,9 +269,6 @@ class LayoutBuilder extends ConstrainedLayoutBuilder<BoxConstraints> {
     required super.builder,
   }) : assert(builder != null);
 
-  @override
-  LayoutWidgetBuilder get builder => super.builder;
```

Comment:
Removing this override removes some information. The super-class types the return type of this property as

```
Widget Function(BuildContext, ConstraintType)
```

This override types the return type (if you resolve the typedef) as

```
Widget Function(BuildContext context, BoxConstraints constraints)
```


Removing this line removes the information that this subclass will always provide the callback with the more specific `BoxConstraints` instead of just the more generic `ConstraintType`.

I think we should keep this override. (sorry, I didn't catch this in my initial review)

Reply:
(This pattern is also used here, but the lint didn't seem to have caught it there?: https://github.com/flutter/flutter/blob/7ab2bf8ff708c9a4c384ebba531e647298961e60/packages/flutter/lib/src/widgets/sliver_layout_builder.dart#L39)

Reply:
Interesting!  It looks like `unnecessary_overrides` has a (new) false positive in this case.

Reply:
I don't think that's true. The new signature is implied by `extends ConstrainedLayoutBuilder<BoxConstraints>`. That should be reflected in documentation, static analysis, etc. all without the explicit override.

Reply:
You're right, @srawlins! I take back my comment. Removing the override should be totally fine here.

Why does the same not apply to the other here instance, though: https://github.com/flutter/flutter/blob/7ab2bf8ff708c9a4c384ebba531e647298961e60/packages/flutter/lib/src/widgets/sliver_layout_builder.dart#L39

I would expect the lint would want that removed as well?

Reply:
That override has a new doc comment, which is a valid reason to override.

Reply:
Makes sense. @pq Since the comment is very similar to the super-class, would you mind just removing it and the override? Just so that these two classes stay symmetric?

Reply:
Sure thing.

Reply:
Thanks for the clarification @srawlins!